editor.planx.uk/src/@planx/components/MapAndLabel/Public/index.tsx

```
@@ -0,0 +1,198 @@
+              {...a11yProps(i)}
+            />
+          ))}
+        </Tabs>
+        {features.map((feature, i) => (
+          <TabPanel
+            key={`tabpanel-${i}`}
+            value={feature.properties?.label}
+            sx={{ width: "100%" }}
+          >
+            <Typography component="h2" variant="h3">
+              {`${schema.type} ${feature.properties?.label}`}
+            </Typography>
+            <Typography variant="body2" gutterBottom>
+              {`${feature.geometry.type}`}
+              {feature.geometry.type === "Point"
+                ? ` (${feature.geometry.coordinates.map((coord) =>
+                    coord.toFixed(5),
+                  )})`
+                : ` (area ${
+                    feature.properties?.["area.squareMetres"] || 0
+                  } m²)`}
+            </Typography>
+            <SchemaFields
+              schema={schema}
+              activeIndex={activeIndex}
+              formik={formik}
+            />
+          </TabPanel>
+        ))}
+      </TabContext>
+    </Box>
+  );
+};
+
+const Root = () => {
+  const { validateAndSubmitForm, mapAndLabelProps } = useMapAndLabelContext();
+  const {
+    title,
+    description,
+    info,
+    policyRef,
+    howMeasured,
+    drawColor,
+    drawType,
+    schemaName,
+  } = mapAndLabelProps;
+
+  const teamSettings = useStore.getState().teamSettings;
+  const passport = useStore((state) => state.computePassport());
+
+  const [features, setFeatures] = useState<Feature[] | undefined>(undefined);
+
+  useEffect(() => {
+    const geojsonChangeHandler = ({ detail: geojson }: any) => {
+      if (geojson["EPSG:3857"]?.features) {
+        setFeatures(geojson["EPSG:3857"].features);
+      } else {
+        // if the user clicks 'reset' on the map, geojson will be empty object, so set features to undefined
+        setFeatures(undefined);
+      }
+    };
+
+    const map: any = document.getElementById("map-and-label-map");
```

```suggestion
    const map: HTMLDivElement | null = document.getElementById("map-and-label-map");
```

---

And then maybe throw an exception if no map found in order to type narrow within this function?